test/test-ip6-addr.c

```diff
@@ -139,3 +139,14 @@ TEST_IMPL(ip6_pton) {
 
 #undef GOOD_ADDR_LIST
 #undef BAD_ADDR_LIST
+
+#ifndef _WIN32
+TEST_IMPL(ip6_invalid_interface) {
+  struct sockaddr_in6 s;
+  int r;
+
+  r = uv_ip6_addr("::0%bad", 0, &s);
+  ASSERT(r == UV_ENXIO);
```

Comment:
I don't know if ENXIO is the only error that if_nametoindex() returns.  It is usually a wrapper around netlink or `ioctl(SIOCGIFINDEX)`, it might return different errors when no network is configured, for example.